### ISSUE TITLE
Why setting policy.train() before collect samples in offline_trainer?

### ISSUE THREAD

**Issue by GIS-PuppetMaster** [thu-ml/tianshou#203]
- [ ] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [ ] RL algorithm bug
    + [ ] documentation request (i.e. "X is missing from the documentation.")
    + [ ] new feature request
- [x] I have visited the [source website](https://github.com/thu-ml/tianshou/)
- [x] I have searched through the [issue tracker](https://github.com/thu-ml/tianshou/issues) for duplicates
- [ ] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import tianshou, torch, sys
  print(tianshou.__version__, torch.__version__, sys.version, sys.platform)
  ```
I have asked a Batchnorm question earlier, after reading pytorch doc, I think it's more reasonable to set policy.eval() rather than policy.train() before collect samples.
```python
for epoch in range(1, 1 + max_epoch):
        # train
        policy.train()
        if train_fn:
            train_fn(epoch)
        with tqdm.tqdm(total=step_per_epoch, desc=f'Epoch #{epoch}',
                       **tqdm_config) as t:
            while t.n < t.total:
                result = train_collector.collect(n_step=collect_per_step)
                data = {}
                if test_in_train and stop_fn and stop_fn(result['rew']):
                    test_result = test_episode(
                        policy, test_collector, test_fn,
                        epoch, episode_per_test)
```
Also, I think it should be set to eval mode before test, incase someone using BatchNorm layer or dropout.

**Comment by Trinkle23897** [comment]
> Also, I think it should be set to eval mode before test, incase someone using BatchNorm layer or dropout.

This has been set in `test_episode` under `tianshou/trainer/utils.py`



> I have asked a Batchnorm question earlier, after reading pytorch doc, I think it's more reasonable to set policy.eval() rather than policy.train() before collect samples.

Indeed, ~~off-policy~~ both trainer should do this ~~, but how about PPO with batch-norm?~~
I'll make a pr to fix it.

**Comment by GIS-PuppetMaster** [comment]
> > Also, I think it should be set to eval mode before test, incase someone using BatchNorm layer or dropout.
> 
> This has been set in `test_episode` under `tianshou/trainer/utils.py`
> 
> > I have asked a Batchnorm question earlier, after reading pytorch doc, I think it's more reasonable to set policy.eval() rather than policy.train() before collect samples.
> 
> Indeed, ~off-policy~ both trainer should do this ~, but how about PPO with batch-norm?~
> I'll make a pr to fix it.

Thanks for replying, and I still have a problem. What is the `self._action_bias` for in DDPGPolicy code? It seems that the action_bias variable is added to action before applying noise.

**Comment by Trinkle23897** [comment]
> Thanks for replying, and I still have a problem. What is the `self._action_bias` for in DDPGPolicy code? It seems that the action_bias variable is added to action before applying noise.

You have already asked this in #194

**Comment by Trinkle23897** [comment]
@zkx741481546 please have a look at #204 and feel free to give any comments!

**Comment by Trinkle23897** [comment]
I have to revert the code because of the issue mentioned in #211
